packages/react-native/src/modules/push-notifications/ExpoPushNotificationProvider.tsx

```diff
@@ -0,0 +1,236 @@
+// PushNotificationContext.tsx
+import {
+  ChannelData,
+  Message,
+  MessageEngagementStatus,
+} from "@knocklabs/client";
+import { useKnockClient } from "@knocklabs/react-core";
+import Constants from "expo-constants";
+import * as Device from "expo-device";
+import * as Notifications from "expo-notifications";
+import React, { createContext, useContext, useEffect, useState } from "react";
+
+export interface ExpoPushNotificationContextType {
+  expoPushToken: string | null;
+  registerForPushNotifications: () => Promise<void>;
+  onNotificationReceived: (
+    handler: (notification: Notifications.Notification) => void,
+  ) => void;
+  onNotificationTapped: (
+    handler: (response: Notifications.NotificationResponse) => void,
+  ) => void;
+}
+
+Notifications.setNotificationHandler({
+  handleNotification: async () => {
+    return {
+      shouldShowAlert: true,
+      shouldPlaySound: true,
+      shouldSetBadge: true,
+    };
+  },
+});
+
+const ExpoPushNotificationContext = createContext<
+  ExpoPushNotificationContextType | undefined
+>(undefined);
+
+export interface ExpoPushNotificationProviderProps {
+  knockExpoChannelId: string;
+  children?: React.ReactElement;
+}
+
+async function requestPermissionAndGetPushToken(): Promise<Notifications.ExpoPushToken | null> {
+  // Check for device support
+  if (!Device.isDevice) {
+    console.log("Must use physical device for Push Notifications");
+    return null;
+  }
+
+  // Check for permissions
+  const { status: existingStatus } = await Notifications.getPermissionsAsync();
+  let finalStatus = existingStatus;
+  if (existingStatus !== "granted") {
+    const { status } = await Notifications.requestPermissionsAsync();
+    finalStatus = status;
+  }
+
+  if (finalStatus !== "granted") {
+    console.log(
+      "Failed to get push token for push notification! User has not granted push notification permissions on device.",
+    );
+    return null;
+  }
+
+  try {
+    if (
+      !Constants.expoConfig ||
+      !Constants.expoConfig.extra ||
+      !Constants.expoConfig.extra.eas
+    ) {
+      console.error("Project ID is not defined in the app configuration.");
+      return null;
+    }
+    const token = await Notifications.getExpoPushTokenAsync({
+      projectId: Constants.expoConfig.extra.eas.projectId,
+    });
+    return token;
+  } catch (error) {
+    console.error("Error getting a push token", error);
+    return null;
+  }
+}
+
+export const ExpoPushNotificationProvider: React.FC<
```

Comment:
Should we include Knock in this provider name like we do with the other providers, including the Slack provider? Since Expo is another organization, the extra namespacing is important imo.